### ISSUE TITLE
Some approved pathways are missing from download archives

### ISSUE THREAD

**Issue by khanspers** [wikipathways/wikipathways.org#72]
There is a difference in total number of approved pathways in the downloadable zip archives (1570) and what is returned by web services/R (1596). After some digging, it looks like some approved pathways are somehow not included in the archives (they are listed in the gmt). Full list below (species code in parenthesis), in case it's relevant. Some of these pathways have WPIDs that are close, so maybe there was a glitch in the release mechanism at certain times? 

Note: I realize the web service returns the daily list, so the difference of 26 includes 5 pathways that are newly tagged (not listed below). Also includes one pathway for Beta vulgaris species for which there is no data archive at all (not listed below). 

- [x] WP2826	Cocaine metabolism (Hs)
- [x] WP2849	Hematopoietic Stem Cell Diff (Hs)
- [x] WP2901	Gonadotropin-releasing hormone (GnRH) signaling (Bt)
- [x] WP2902	p53 Signaling (Mm)
- [x] WP2904	Mir302-367 Promoting Cardiomyocyte Proliferation (Mm)
- [x] WP2910	Mecp2 and Associated Rett Syndrome (Mm)
- [x] WP2944	Cytokinin signaling (At)
- [x] WP332	RuMP cycle, containing "Oxidative Branch of the Pentose Phosphate Pathway" and formaldehyde assimilation (Sc)
- [x] WP3585	Cytosine methylation (Hs)
- [x] WP3591	Sleep regulation (Hs)
- [x] WP3611	Photodynamic therapy-induced AP-1 survival signaling (Hs)
- [x] WP3627	Gut-Liver Indole Metabolism (Hs)
- [x] WP3894	Fructose Metabolism in Proximal Tubules (Rn)
- [x] WP3963	Mevalonate pathway (Hs)
- [x] WP3972	PDGFR-beta pathway (Hs)
- [x] WP3979	mir-193a and MVP in colon cancer metastasis (Hs)
- [x] WP3981	miRNA regulation of prostate cancer signaling pathways (Hs)
- [x] WP4205	MET in type 1 papillary renal cell carcinoma (Hs)
- [x] WP4206	Hereditary leiomyomatosis and renal cell carcinoma pathway (Hs)
- [x] WP4210	Tryptophan catabolism leading to NAD+ production (Hs)

**Comment by mkutmon** [comment]
Would it be worth it to retag those pathways and see if they are in the new released collection tomorrow? I'll do that. I can't think of another reason why they are not included.

**Comment by mkutmon** [comment]
okay, so I retagged all of them - let's check tomorrow!

**Comment by khanspers** [comment]
Just checked and the pathways are still missing.... total count is 1573, but should be 1594 based on web services.

**Comment by khanspers** [comment]
This is still a a problem. Tagged gpmls are still not in downloaded archives.

**Comment by egonw** [comment]
I note that some have "reverts", but not all of them. Can we easily rule out that "reverts" are one possible cause?

@hexmode, do you know if there is an easy MediaWiki way of listing all Wiki pages with revert changes?

**Comment by hexmode** [comment]
Egon Willighagen writes:

> @hexmode, do you know if there is an easy MediaWiki way of listing all Wiki pages with revert changes?

Not offhand.  I'll have to look at the code and come back.

**Comment by khanspers** [comment]
But many pathways have been reverted that are in the download archive, so that's likely not it. 

@mkutmon Does the data release building script produce any kind of log?
The only thing I can think of in terms of debugging would be to upload one of the missing gpmls as a new pathway (right before the next release build), tag it and see if it gets included. If the answer is no, then points to the gpml.

**Comment by khanspers** [comment]
Latest updates/information:
- missing gpmls are in the gmt and pdf and available from web services (via tag) => this is a problem with the internal tagging
- Different number of pathways in gmt / web services / WP stats

=> close for now since this should all be solved with new GPML repo

**Comment by egonw** [comment]
@khanspers, this was a good catch to link Freddie's problem to this one. While exploring her problem, I found that WP4932 had been deleted by @andrawaag on June 0th 2020:

```
10:23, 9 June 2020 Andra (Talk | contribs | block) deleted "Pathway:WP4932" ‎ (content was: '<?xml version="1.0" encoding="UTF-8"?> <Pathway xmlns="http://pathvisio.org/GPML/2013a" Name="Apoptosis thesis" Organism="Homo sapiens"> <Graphics BoardWidth="3...' (and the only contributor was 'Debxgsrt')) (restore)
```

And nine days later, WP4932 as pathways was reused when Freddie started her new pathway.

I just checked WP2826 from this report, and it was deleted on June 16 2015 and a day later the WP2826 identifier was reused when @andrawaag created a new pathway.

So, that supports the idea that the two problems are connected: deleted pathways keep the flag "deleted" and now showing up when asking in the PHP code for pathways with a tag. This I confirmed this in the `MetaTag::getPagesForTag` method, by removing the last conditional in:

```php
if(!$title || $title->isRedirect() || $title->isDeleted()) {
    continue; //Skip redirects and deleted
}
```

When I removed the `$title->isDeleted())` the WP4932 from Freddie did show up in the list for the tag.

So, the solution I am not clear about at this moment, and should likely include two parts:

1. make an overview of problematic pathways (ideally, by a well-crafted SQL query) and remove the spurious 'deleted' tags in the database
2. fix the pathway creation code to reset the 'deleted' tag when it "reuses" a pathway ID

**Comment by egonw** [comment]
I may have found the solution to remove the spurious tasks

**Comment by egonw** [comment]
The following SQL lists all pathways we may need to check, but includes false positives too:

```sql
SELECT DISTINCT page_title, MAX(page_touched) last_touched, MAX(ar_timestamp) last_deleted FROM archive, page WHERE page_title = ar_title AND page_title LIKE 'WP%' GROUP BY page_title HAVING last_touched > last_deleted;
```

**Comment by khanspers** [comment]
Great progress! If you can get a list of WPIDs from your SQL query, I can check them. 

Also, I thought the reusing of deleted WPIDs was fixed a long time ago, so it should not be doing that anymore?

**Comment by egonw** [comment]
Email of 169 possible matches via Slack...

> Also, I thought the reusing of deleted WPIDs was fixed a long time ago, so it should not be doing that anymore?

Oh, that's still happening... a recent example is WP5052 (one of the two Freddie found).

**Comment by khanspers** [comment]
Hi all,

After the mystery entry on the agenda for today's call, I looked at this
again. A few weeks ago Egon came up with a solution for this, to manually
remove the spurious 'deleted' tags in the database. So based on the list of
169 pathways with a deletion tag in the database that Egon sent me (from
SQL query), I created a list of "true positives": pathways that exist and
should NOT have the deleted tag. Attaching the list here, see "True
positives" tab.

Hopefully someone who has access to the database can make these updates?

Kristina
